### ISSUE TITLE
Github actions provides inconsistent macos-13 runner versions

### ISSUE THREAD

**Issue by marcindulak** [actions/runner-images#9595]
### Description

Github actions provides inconsistent macos-13 runner versions.

This is similar to https://github.com/actions/runner-images/issues/9568, which reports inconsistent macos-14 runner versions.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [X] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

https://github.com/marcindulak/github-actions-runner-images-9595/actions/runs/8494165449/job/23268975511

### Is it regression?

No

### Expected behavior

The "Version: YYYYMMDD.N" is consistent across jobs

```
Runner Image
  Image: macos-13
  Version: YYYYMMDD.N
```

### Actual behavior

Two out of 10 jobs use

```
Runner Image
  Image: macos-13
  Version: 20240219.1
```

<img width="1073" alt="Screenshot 2024-03-31 at 00 56 56" src="https://github.com/actions/runner-images/assets/3178318/11d6bda8-3b9f-4b7e-8c80-5de17ca69d63">

The rest uses

```
Runner Image
  Image: macos-13
  Version: 20240327.2
```

<img width="1074" alt="Screenshot 2024-03-31 at 00 55 09" src="https://github.com/actions/runner-images/assets/3178318/474bc6bd-5654-4728-ae01-c4494b89b1aa">

### Repro steps

Trigger https://github.com/marcindulak/github-actions-runner-images-9595 a few times.

**Comment by shamil-mubarakshin** [comment]
Hey @marcindulak. We will take a look

**Comment by shamil-mubarakshin** [comment]
Should be fixed now, with `20240327.2` being consistent
